ext/src/autobahn/encoder/MsgPackEncoderBase.java

```diff
@@ -0,0 +1,89 @@
+package autobahn.encoder;
+
+
+import java.io.IOException;
+
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyObject;
+import org.jruby.RubyHash;
+import org.jruby.RubyString;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.jruby.runtime.ObjectAllocator;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.anno.JRubyClass;
+import org.jruby.util.ByteList;
+
+import static org.jruby.runtime.Visibility.*;
+
+import com.headius.jruby.lz4.vendor.net.jpountz.lz4.LZ4Compressor;
+import com.headius.jruby.lz4.vendor.net.jpountz.lz4.LZ4Decompressor;
+import com.headius.jruby.lz4.vendor.net.jpountz.lz4.LZ4Factory;
+
+import org.msgpack.jruby.Encoder;
+import org.msgpack.jruby.Decoder;
+
+
+abstract public class MsgPackEncoderBase extends RubyObject {
+  private static final RubyString CONTENT_TYPE = Ruby.getGlobalRuntime().newString("application/msgpack");
```

Comment:
I realize that this isn't new, and also that it doesn't really matter for our use cases, but technically, this string should be allocated in the same runtime as the encoder is allocated.